src/frontends/jax/src/translate_session.cpp

```diff
@@ -0,0 +1,364 @@
+#include "openvino/util/log.hpp"
+#include "place.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace jax {
+
+using namespace ov::op;
+
+TranslateSession::TranslateSession(const ov::frontend::InputModel::Ptr& input_model,
+                                   const std::map<std::string, CreatorFunction>& translator_map,
+                                   const std::shared_ptr<TelemetryExtension>& telemetry)
+    : m_input_model(input_model),
+      m_translator_map(translator_map),
+      m_telemetry(telemetry),
+      m_ov_model(nullptr) {}
+
+TranslateSession::~TranslateSession() {
+    if (m_telemetry) {
+        // Send statistics
+        for (const auto& op : m_op_statistics) {
+            m_telemetry->send_event("op_count", "jax_" + op.first, static_cast<int>(op.second));
+        }
+    }
+}
+
+std::shared_ptr<ov::Model> TranslateSession::get_converted_model() {
+    if (m_ov_model) {
+        return m_ov_model;
+    }
+    m_ov_model = translate_graph(m_input_model);
+    return m_ov_model;
+}
+
+std::shared_ptr<ov::Model> TranslateSession::translate_graph(const ov::frontend::InputModel::Ptr& input_model) {
+    auto jax_model = std::dynamic_pointer_cast<jax::InputModel>(input_model);
+    FRONT_END_GENERAL_CHECK(jax_model != nullptr, "Invalid input model");
+    auto model = convert_jax_model(jax_model->m_model_decoder, {}, jax_model);
+    // First delete tensor indexes from outputs then resolve input names,
+    // otherwise Parameter->Result will fail
+    for (auto& result : model->get_results()) {
+        auto tensor_desc = result->input_value(0);
+        auto names = tensor_desc.get_names();
+        if (!names.empty()) {
+            auto tensor_idx = decode_tensor_name(tensor_desc);
+            if (names.erase(std::to_string(tensor_idx))) {
+                tensor_desc.set_names(names);
+            }
+        }
+    }
+    // Set input tensor names to be equal to signature name saved in friendly name
+    for (auto& param : model->get_parameters()) {
+        if (param->get_friendly_name() != param->get_name()) {
+            // get_name is autogenerated name, we need to make sure that this
+            // parameter was named by frontend
+            param->output(0).set_names({param->get_friendly_name()});
+        }
+    }
+    return model;
+}
+
+std::shared_ptr<Model> TranslateSession::convert_jax_model(std::shared_ptr<JaxDecoder> jax_model,
+                                                           const TensorMap& external_tensor_map,
+                                                           const std::shared_ptr<jax::InputModel>& input_model) {
+    std::shared_ptr<Model> resulting_model;  // define here to make a conversion in a nested scope
+    {
+        auto parameters = std::make_shared<ParameterVector>();
+        auto tensor_map = std::make_shared<TensorMap>();  // tensor map of the current context
+        auto mutated_tensors = std::make_shared<std::set<size_t>>();
```

Comment:
I would recommend removing mutated tensor logic for now. It may be implemented dufferently in JAX